ShelterTechSF/askdarcel-web
# 1135 Update to Node 18 and Webpack 5

Ticket: https://github.com/ShelterTechSF/askdarcel-web/issues/1135
This pull request updates node from v10 to v18. It also updates a number of other dependencies to resolve peer dependency conflicts and bring them in line with the latest node version. Some other changes include:

- Migrating to Cypress 10
- Migrating to Webpack 5
- Migrating to eslint 7

This resulted in/required a number of code changes. Most significantly, updating eslint yielded quite a few lint errors. Running `npm run lint:fix` made changes to many files, and many more lint errors required manual fixes. I've annotated the most salient ones in the comments.

To test this PR:

- Pull down this branch
- Check if you have node 18 installed: `nvm ls`
- Install it if necessary `nvm install 18`
- These changes affected dependencies that I recommend starting with fresh `node_modules` and `package-lock.json`:
```
rm -rf node_modules/
rm package-lock.json
npm install
```
- Run the API locally
- Run this branch locally
- Visit the site and test manually
- Run the cypress tests: `npx cypress open`
- Run the linter: `npm run lint`
